### PR TITLE
Update edgeagentuser and edgehubuser to nologin

### PR DESCRIPTION
**Pull request by yophilav**
Update edgeagentuser and edgehubuser to nologin
Address https://github.com/Azure/iotedge/issues/6747

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

**Comment by arsing** [comment]
https://github.com/Azure/iotedge/blob/main/edgelet/contrib/centos/aziot-edge.spec and https://github.com/Azure/iotedge/blob/main/edgelet/contrib/enterprise-linux/aziot-edge.spec also need to be updated.

Also https://github.com/Azure/iotedge/blob/main/builds/mariner/SPECS/aziot-edge/aziot-edge.spec seems to not be creating the EH user for some reason.

Also also, when did we start creating these users on the host and why? I see https://github.com/Azure/iotedge/commit/388ec1a341cf18fce3379c6750e62591963c8624 made it so. @gordonwang0 ?

**Review by gordonwang0 — APPROVED** [review]

File: edgelet/contrib/centos/aziot-edge.spec [hunk 1]
```diff
@@ -97,12 +97,12 @@ fi
 
 # Create an edgeagentuser and add it to iotedge group
 if ! /usr/bin/getent passwd %{iotedge_agent_user} >/dev/null; then
-    %{_sbindir}/useradd -r -g %{iotedge_group} -c "edgeAgent user" -s /bin/sh -u %{iotedge_agent_uid} %{iotedge_agent_user} || true
+    %{_sbindir}/useradd -r -g %{iotedge_group} -c "edgeAgent user" -s /bin/nologin -u %{iotedge_agent_uid} %{iotedge_agent_user} || true
```
Comment (arsing) [inline edgelet/contrib/centos/aziot-edge.spec:100]: Are you sure you have the right path? A CentOS 7 container has `/sbin/nologin`, not `/bin/login`.

Comment (yophilav) [reply edgelet/contrib/centos/aziot-edge.spec:100]: Yeah, I think so. I used what's in the file i.e.
https://github.com/Azure/iotedge/blob/release/1.4/edgelet/contrib/centos/aziot-edge.spec#L85

Comment (arsing) [reply edgelet/contrib/centos/aziot-edge.spec:100]: That file is also wrong then.

Comment (yophilav) [reply edgelet/contrib/centos/aziot-edge.spec:100]: Oh ok. I'll use the `sbin`.

**Review by arsing — APPROVED** [review]

**Pull request merged**